common/predictive-text/worker/model-compositor.ts

```diff
@@ -438,19 +451,33 @@ class ModelCompositor {
       deleteLeft: insertedLength
     };
 
+    let postCaretReversionTransform: Transform = {
+      insert: deletedRightChars,
+      deleteLeft: 0
+    }
+
     // Step 2:  building the proper 'displayAs' string for the Reversion
     let postContext = context;
     if(postTransform) {
       // The code above restores the state to the context at the time the `Suggestion` was created.
       // `postTransform` handles any missing context that came later.
       reversionTransform = models.buildMergedTransform(reversionTransform, postTransform);
+      reversionTransform = models.buildMergedTransform(reversionTransform, postCaretReversionTransform);
 
       // Now that we've built the reversion based upon the Suggestion's original context,
       // we manipulate it in order to get a proper 'displayAs' string.
       postContext = models.applyTransform(postTransform, postContext);
     }
 
-    let revertedPrefix = this.wordbreak(postContext);
+    let revertedPrefix: string;
+    let postContextTokenization = this.tokenize(postContext);
+    if(postContextTokenization) {
+      // Handles display string for reversions triggered by accepting a suggestion mid-token.
+      revertedPrefix = postContextTokenization.left[postContextTokenization.left.length-1];
+      revertedPrefix += postContextTokenization.caretSplitsToken ? postContextTokenization.right[0] : '';
```

Comment:
Can we be certain that `postContextTokenization.right` always contains at least one element?

Reply:
If `postContextTokenization.caretSplitsToken == true`, yes.  Otherwise, no.